storage/innobase/handler/ha_innodb.cc

```diff
@@ -16887,6 +16912,71 @@ innodb_status_output_update(
 	os_event_set(lock_sys->timeout_event);
 }
 
+/*************************************************************//**
+Empty free list algorithm.
+Checks if buffer pool is enough to enable backoff algorithm.
+@return	true if it's not possible to enable backoff. */
+static
+bool
+innodb_empty_free_list_algorithm_check(
+	ulong 		algorithm,		/*!< in: desired algorithm
+						from srv_empty_free_list_t */
+	long long	pages_in_buffer_pool)	/*!< in: total number of
+						pages inside buffer pool */
+{
+	return(pages_in_buffer_pool < BUF_LRU_MIN_LEN * (4 + 1)
+			&& algorithm == SRV_EMPTY_FREE_LIST_BACKOFF);
+}
+
+/*************************************************************//**
+Empty free list algorithm. This function is registered as
+a callback with MySQL. 
+@return	0 for valid algorithm */
+static
+int
+innodb_srv_empty_free_list_algorithm_validate(
+/*===========================*/
+	THD*				thd,	/*!< in: thread handle */
+	struct st_mysql_sys_var*	var,	/*!< in: pointer to system
+						variable */
+	void*				save,	/*!< out: immediate result
+						for update function */
+	struct st_mysql_value*		value)	/*!< in: incoming string */
+{
+	const char*	algorithm_name;
+	char		buff[STRING_BUFFER_USUAL_SIZE];
+	int		len = sizeof(buff);
+	ulint	algo;
+	srv_empty_free_list_t 		algorithm;
+
+	algorithm_name = value->val_str(value, buff, &len);
+
+	if (!algorithm_name) {
+		return(1);
+	}
+
+	for (algo = 0; algo < array_elements(
+				innodb_empty_free_list_algorithm_names
+				) - 1;
+			algo++) {
+		if (!innobase_strcasecmp(
+				algorithm_name,
+				innodb_empty_free_list_algorithm_names[algo]))
+			break;
+	}
+
+	if (algo == array_elements( innodb_empty_free_list_algorithm_names) - 1)
+		return(1);
+
+	algorithm = static_cast<srv_empty_free_list_t>(algo);
+	if (innodb_empty_free_list_algorithm_check(algorithm,
+				innobase_buffer_pool_size / srv_page_size))
+		return(1);
```

Comment:
I am starting to think that maybe sql_print_warning is required here why the variable change was forbidden?